pkg/health/health.go

```diff
@@ -0,0 +1,46 @@
+package health
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/spiffe-helper/pkg/sidecar"
+)
+
+type CheckConfig struct {
+	ListenerEnabled bool   `hcl:"listener_enabled"`
+	BindPort        int    `hcl:"bind_port"`
+	HealthPath      string `hcl:"health_path"`
+}
+
+func StartHealthServer(daemonMode bool, healthCheckConfig CheckConfig, log logrus.FieldLogger, sidecar *sidecar.Sidecar) error {
+	if daemonMode && healthCheckConfig.ListenerEnabled {
```

Comment:
Can we move this check to main before calling this function? That'll reduce the indentation a little bit.